src/components/Slot/Slot.tsx

```diff
@@ -31,6 +39,7 @@ const Slot: CreateComponentReturnType<SlotProps> & {
     ...commonUIComponentPropTypes,
     ...contentComponentPropsTypes,
     ...childrenComponentPropTypes,
+    innerRef: PropTypes.func,
```

Comment:
do we really need it now? why not to use `Ref`?

Reply:
we debated this many times, and we agreed for now to have it as handled prop to avoid console warning. It is populated with the Ref (via the HOC you suggested) but it still needs to be handled for that warning reason.